src/transaction/log_page_buffer.c

```diff
@@ -726,35 +527,58 @@ logpb_initialize_pool (THREAD_ENTRY * thread_p)
 
   assert (LOG_CS_OWN_WRITE_MODE (thread_p));
 
-  if (log_Pb.pool != NULL)
+  if (log_Pb.buffers != NULL)
     {
       logpb_finalize_pool (thread_p);
     }
 
-  assert (log_Pb.pool == NULL && log_Pb.poolarea == NULL && log_Pb.ht == NULL);
+  assert (log_Pb.buffers == NULL);
 
-  log_Pb.num_buffers = 0;
-  log_Pb.pool = NULL;
-  log_Pb.poolarea = NULL;
+  log_Pb.buffers = NULL;
 
   /* 
    * Create an area to keep the number of desired buffers
    */
-  error_code = logpb_expand_pool (thread_p, -1);
-  if (error_code != NO_ERROR)
+  log_Pb.num_buffers = prm_get_integer_value (PRM_ID_LOG_NBUFFERS);
+
+  size = ((log_Pb.num_buffers * sizeof (LOG_BUFFER)) + sizeof (LOG_BUFAREA));
+
+
+
+  /* allocate a pointer array to point to each buffer */
+  log_Pb.buffers = (LOG_BUFFER *) malloc ((size_t) ((size_t) log_Pb.num_buffers * sizeof (*log_Pb.buffers)));
+  if (log_Pb.buffers == NULL)
     {
-      goto error;
+
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1,
+	      log_Pb.num_buffers * sizeof (*log_Pb.buffers));
+      return ER_OUT_OF_VIRTUAL_MEMORY;
     }
 
-  log_Pb.ht = mht_create ("Log buffer pool hash table", log_Pb.num_buffers, mht_logpageidhash,
-			  mht_compare_logpageids_are_equal);
-  if (log_Pb.ht == NULL)
+  log_Pb.pages_area = (LOG_PAGE *) malloc ((size_t) ((size_t) log_Pb.num_buffers * (LOG_PAGESIZE)));
+  if (log_Pb.pages_area == NULL)
     {
-      error_code = ER_OUT_OF_VIRTUAL_MEMORY;
-      goto error;
+
+      free_and_init (log_Pb.buffers);
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1,
+	      log_Pb.num_buffers * sizeof (*log_Pb.buffers));
```

Comment:
@funkydvd 

Please correct the size parameter of the error message. It looks like copy-paste error. :)